sympy/physics/vector/frame.py

```diff
@@ -762,6 +762,44 @@ def orient_explicit(self, parent, dcm):
         parent._ang_vel_dict.update({self: -wvec})
         self._var_dict = {}
 
+
+    def orient_dcm(self, parent, dcm):
+        """Sets the orientation of this reference frame relative to a parent
+        reference frame using a direction cosine matrix that describes the rotation
+        from the parent to the child.
+
+        Parameters
+        ==========
+
+        parent : ReferenceFrame
+            Reference frame that this reference frame will be rotated relative
+            to.
+        dcm : Matrix, shape(3, 3)
+            Direction cosine matrix that specifies the relative rotation between
+            the two reference frames. The DCM should describe the rotation from the
+            parent to the child reference frame.
+
```

Comment:
I would remove the examples from the `orient_explicit` method and move them here. Note that you will have to update them a bit.

Reply:
It would be better to leave the docstring in, maybe just remove the examples and clarify the rotation direction, i.e. child to parent. Instead you can add `:exclude-members: orient_explicit` in the [doc/.../physics/vector/classes.rst](https://github.com/sympy/sympy/blame/master/doc/src/modules/physics/vector/api/classes.rst#L9).

Reply:
``` 
 def orient_dcm(self, parent, dcm):
        """Sets the orientation of this reference frame relative to a parent
        reference frame by explicitly setting the direction cosine matrix.

        Parameters
        ==========

        parent : ReferenceFrame
            Reference frame that this reference frame will be rotated relative
            to.
        dcm : Matrix, shape(3, 3)
            Direction cosine matrix that specifies the relative rotation
            between the two reference frames.

        rotation matrix : it depends on the dcm matrix
            The direction of vector rotation is counterclockwise if θ is positive (e.g. 90), and clockwise if θ is negative (e.g. −90).
            
        """

        _check_frame(parent)
        # amounts must be a Matrix type object
        # (e.g. sympy.matrices.dense.MutableDenseMatrix).
        if not isinstance(dcm, MatrixBase):
            raise TypeError("Amounts must be a SymPy Matrix type object.")

        

        self._dcm(parent, dcm.T)

        wvec = self._w_diff_dcm(parent)
        self._ang_vel_dict.update({parent: wvec})
        parent._ang_vel_dict.update({self: -wvec})
        self._var_dict = {}
 ```
 As you mention in [here](https://github.com/sympy/sympy/issues/24764#issuecomment-1445953996) Complete the task :
* change name from `orient_explicit` to `orient_dcm`
* removing docstring and adding rotation matrix direction
*  Is it fine?

Reply:
> * change name from `orient_explicit` to `orient_dcm`

Yes, the user should be pointed to `orient_dcm` as default. So changing the name and using the transpose sounds good. However, we should keep backwards compatibility so `orient_explicit` should also be implemented to just call `self.orient_dcm(parent, dcm.T)`. Make sure to also have a clear docstring for this one.

> * removing docstring and adding rotation matrix direction

Not entirely sure what you mean, but both `orient_dcm` and `orient_explicit` should have a clear docstring that also specifies what direction the DCM should be defined in.

Additionally, we want to exclude `orient_explicit` from the online documentation to point users to `orient_dcm`. This can be done by adding :exclude-members: orient_explicit in the [doc/.../physics/vector/classes.rst](https://github.com/sympy/sympy/blame/master/doc/src/modules/physics/vector/api/classes.rst#L9).

Reply:
>  * removing docstring and adding rotation matrix direction

DCM rotation depends on the angle Theta. If I am making any mistakes, please let me know. Check this [reference](https://testbook.com/maths/rotation-matrix) .

Reply:
> * removing docstring and adding rotation matrix direction

dcm rotation matrix direction depends on the cosine of theta. This is the [reference](https://testbook.com/maths/rotation-matrix) I found. Please let me know if I made a mistake. 
Is this correct? `The direction of vector rotation is counterclockwise if θ is positive (e.g. 90), and clockwise if θ is negative (e.g. −90).`

Reply:
> Is this correct? `The direction of vector rotation is counterclockwise if θ is positive (e.g. 90), and clockwise if θ is negative (e.g. −90).`

No, this is not correct. You may want to refer to [this explanation](https://moorepants.github.io/learn-multibody-dynamics/orientation.html#direction-cosine-matrices) by moorepants.

Reply:
I apologize for the spam, but are we discussing this rotation matrix? ``B.dcm(A)``  it means the "direction cosine
        matrix of B rotated relative to A"`

Reply:
No problem, feel free to ask questions. Yes, `B.dcm(A)` gives the orientation of `B` with respect to `A`  (as mentioned [in this part](https://moorepants.github.io/learn-multibody-dynamics/orientation.html#sympy-mechanics)).